crates/editor/src/element.rs

```diff
@@ -2620,6 +2485,156 @@ impl EditorElement {
+            editor.scrollbar_marker_state.dirty = false;
+            editor.scrollbar_marker_state.pending_refresh =
+                Some(cx.spawn(|editor, mut cx| async move {
+                    let scrollbar_size = scrollbar_layout.hitbox.size;
+                    let scrollbar_markers = cx
+                        .background_executor()
+                        .spawn(async move {
+                            let mut marker_quads = Vec::new();
+
+                            if scrollbar_settings.git_diff {
+                                let marker_row_ranges = snapshot
+                                    .buffer_snapshot
+                                    .git_diff_hunks_in_range(0..max_row)
+                                    .map(|hunk| {
+                                        let start_display_row =
+                                            Point::new(hunk.associated_range.start, 0)
+                                                .to_display_point(&snapshot.display_snapshot)
+                                                .row();
+                                        let mut end_display_row =
+                                            Point::new(hunk.associated_range.end, 0)
+                                                .to_display_point(&snapshot.display_snapshot)
+                                                .row();
+                                        if end_display_row != start_display_row {
+                                            end_display_row -= 1;
+                                        }
+                                        let color = match hunk.status() {
+                                            DiffHunkStatus::Added => theme.status().created,
+                                            DiffHunkStatus::Modified => theme.status().modified,
+                                            DiffHunkStatus::Removed => theme.status().deleted,
+                                        };
+                                        ColoredRange {
+                                            start: start_display_row,
+                                            end: end_display_row,
+                                            color,
+                                        }
+                                    });
+
+                                marker_quads.extend(
+                                    scrollbar_layout.marker_quads_for_ranges(marker_row_ranges, 0),
+                                );
+                            }
+
+                            for (background_highlight_id, (_, background_ranges)) in
+                                background_highlights.iter()
+                            {
+                                if (*background_highlight_id
+                                    == TypeId::of::<BufferSearchHighlights>()
+                                    && scrollbar_settings.selections)
+                                    || scrollbar_settings.symbols_selections
```

Comment:
@mrnugget, is this expression correct? It means that `symbols_selections = true` forces search highlight markers to be painted even if `selections = false`.

I know that it copies the logic of the previous implementation, but it looked like an error there too becaused it caused search-markers duplication. I always thought that the `selection` setting is for search highlights, and `symbols_selection` is for the `DocumentHighlightRead/Write` highlights.

Am I mistaken?

Reply:
Yeah, we weren't sure about that either. We also suspected that one was a superset of the other and that we did the work twice, but weren't sure what the desired behavior is. You were more involved, I guess, so if you say that this is not correct, then you're probably right.